components/authz-service/engine/opa/policy/authz_v2_test.rego

```diff
@@ -373,3 +373,56 @@ test_authorized_project_deny_real_data {
 
 	actual_projects == set()
 }
+
+test_authorized_project_returns_no_projects_when_all_projects_denied {
+	actual_projects = authorized_project with data.roles.operator.actions as ["y"]
+		 with data.policies.polid as {
+			"members": ["x"],
+			"statements": {
+				"sid-1": {"effect": "allow", "role": "operator", "resources": ["*"], "projects": ["p1", "p3"]},
+				"sid-2": {"effect": "deny", "role": "operator", "resources": ["*"], "projects": [const_all_projects]},
+			},
+		}
+		 with input as {"subjects": ["x"], "action": "y", "resource": "z", "projects": ["p1", "p3", "p5"]}
+
+	actual_projects == set()
+}
+
+test_authorized_project_matches_only_allowed_projects_when_some_projects_denied {
+	actual_projects = authorized_project with data.roles.operator.actions as ["y"]
```

Comment:
I'm still new to rego-- why use the `=` here instead of the `:=`? Are the tests functioning like rules? (Seems like `:=` is reserved for rules . . .)

Reply:
good question. according to [opa's rego guide](https://www.openpolicyagent.org/docs/v0.10.7/how-do-i-write-policies/#the-basics), 
```
There’s one exception: if you use assignment := the compiler will check that the variable you are assigning has not already been used.

s {
    y := 41
    x := 42
    x > y
}
```

Reply:
and yeah, those cases are basically rego rules specifically written for testing. opa has [a good doc](https://www.openpolicyagent.org/docs/v0.10.7/how-do-i-test-policies/) on it

Reply:
so we could use `:=` here?